src/Nox.Types/Types/EncryptedText/EncryptedText.cs

```diff
@@ -1,9 +1,108 @@
+using System;
+using Nox.Types.EncryptionText.EncryptionMethods;
+
 namespace Nox.Types;
 
+/// <summary>
+/// Represents a Nox <see cref="EncryptedText"/> type and value object.
+/// </summary>
+public sealed class EncryptedText : ValueObject<byte[], EncryptedText>
+{
+    public EncryptedText()
+    {
+        Value = new byte[] {};
+    }
+
+    /// <summary>
+    /// Creates an <see cref="EncryptedText"/> from a string using the provided <paramref name="options"/>.
+    /// </summary>
+    /// <param name="value">Plain text to be encrypted.</param>
+    /// <param name="options">Options used for encryption.</param>
+    /// <returns>Encrypted text.</returns>
+    /// <exception cref="TypeValidationException"></exception>
+    public static EncryptedText FromPlainText(string value, EncryptedTextOptions options)
+    {
+        var newObject = new EncryptedText
+        {
+            Value = EncryptText(value, options)
+        };
+
+        var validationResult = newObject.Validate();
+
+        if (!validationResult.IsValid)
+        {
+            throw new TypeValidationException(validationResult.Errors);
+        }
+
+        return newObject;
+    }
+
+    /// <summary>
+    /// Creates an <see cref="EncryptedText"/> from an encrypted data as byte array.
+    /// </summary>
+    /// <param name="value">Encrypted data as byte array.</param>
+    /// <returns></returns>
+    /// <exception cref="TypeValidationException"></exception>
+    public static EncryptedText FromEncryptedText(byte[] value)
```

Comment:
Can we call the method something like "FromEncryptedTextBytes"

Reply:
Method name updated.